src/lib/util/flag-resolver.ts

```diff
@@ -0,0 +1,53 @@
+import {
+    IExperimentalOptions,
+    IExternalFlagResolver,
+    IFlagContext,
+    IFlags,
+    IFlagResolver,
+    IUIFlags,
+} from '../types/experimental';
+
+const DEFAULT_DYNAMIC_FLAGS = ['ENABLE_DARK_MODE_SUPPORT'];
+
+const DEFAULT_EXT_RESOLVER = { isEnabled: () => false };
+
+export default class FlagResolver implements IFlagResolver {
+    private uiFlags: IFlags;
+
+    private experiments: IFlags;
+
+    private dynamicFlags: string[] = [];
+
+    private externalResolver: IExternalFlagResolver;
+
+    constructor(uiFlags: IUIFlags, expOpt: IExperimentalOptions) {
+        this.uiFlags = uiFlags || {};
+        this.experiments = expOpt.flags || {};
+        this.dynamicFlags = [
```

Comment:
Do we need the list of dynamic flags if we always check the static flags first? Seems nice to check static flags first, and then fallback to the resolver (which defaults to false).

Reply:
Could also check the presence of the key instead of its true/false value if we'd like to enforce that some flags never use the resolver (as mentioned), though that seems brittle. Maybe the list is better.

Reply:
> Could also check the presence of the key instead of its true/false value 

I did that first, but it feels a bit strange to always try to resolve something that is a static ui flag that is used for business purposes and not experimental features.